NutriApp/Views/Evaluation/IdealWeight/IdealWeightViewModel.cs

```diff
@@ -1,69 +1,66 @@
-        
-        private bool _hasErrorImcIdeal;
-        public bool HasErrorImcIdeal { get => _hasErrorImcIdeal; set { _hasErrorImcIdeal = value; OnPropertyChanged("HasErrorImcIdeal"); } }
-        
-        public ObservableCollection<Item> ItemImcIdeal { get; set; } 
-        public Command CalculateCommand { get; set; }
-        public Command InfoCommand { get; set; }
-        public IdealWeightViewModel()
+    [ObservableProperty]
+    private bool _canDisplay;
+    [ObservableProperty]
+    private Item _imcIdeal;
+    [ObservableProperty]
+    private string _height;
+    [ObservableProperty]
+    private string _idealWeight;
+    [ObservableProperty]
+    private bool _hasErrorHeight;
+    [ObservableProperty]
+    private bool _hasErrorImcIdeal;
+    public ObservableCollection<Item> ItemImcIdeal { get; set; }
+    public Command InfoCommand { get; set; }
+
+    public IdealWeightViewModel()
+    {
+        InfoCommand = new Command(() => Shell.Current.CurrentPage.ShowPopup(new InfoIdealWeightPopup()));
+        CanDisplay = false;
+        Fill();
+    }
+
+    private void Fill()
+    {
+        ItemImcIdeal = new ObservableCollection<Item>
+        {
+            new() { Id = 22, Nome = "Homem 22 kg/m²" },
+            new() { Id = 21, Nome = "Mulher 21 kg/m²" }
+        };
+    }
+    
+    [RelayCommand]
+    private void Calculate()
+    {
+        if (!Validate())
         {
-            CalculateCommand = new Command(Calculate);
-            InfoCommand = new Command(() => Shell.Current.CurrentPage.ShowPopup(new InfoIdealWeightPopup()));
             CanDisplay = false;
-            Fill();
+            return;
         }
 
-        private void Fill()
+        IdealWeight = EvaluationCalculations
+            .IdealWeight(Convert.ToDouble(ImcIdeal.Id), Utils.Utils.ConvertHeight(Height))
+            .ToString(CultureInfo.InvariantCulture);
+        if (!string.IsNullOrEmpty(IdealWeight))
         {
-            ItemImcIdeal =  new ObservableCollection<Item>
-            {
-                new() {Id = 22, Nome = "Homem 22 kg/m²"},
-                new() {Id = 21, Nome = "Mulher 21 kg/m²"}
-            };
+            CanDisplay = true;
         }
+    }
 
-        private void Calculate()
-        {
-            if (!Validate())
-            {
-                CanDisplay = false;
-                return;
-            }
-
-            IdealWeight = EvaluationCalculations.IdealWeight(Convert.ToDouble(ImcIdeal.Id), Utils.Utils.ConvertHeight(Height)).ToString(CultureInfo.InvariantCulture);
-            if (!string.IsNullOrEmpty(IdealWeight))
-            {
-                CanDisplay = true;
-            }
-        }
-        private bool Validate()
-        {
-            HasErrorHeight = string.IsNullOrWhiteSpace(Height);
-            HasErrorImcIdeal = ImcIdeal == null;
+    private bool Validate()
+    {
+        HasErrorHeight = string.IsNullOrWhiteSpace(Height);
+        HasErrorImcIdeal = ImcIdeal == null;
```

Comment:
```suggestion
        HasErrorImcIdeal = ImcIdeal is null;
```

Como tá fazendo a validação já na atribuição, acho que fica mais legível assim.